app/controllers/api/v2/template_combinations_controller.rb

```diff
@@ -36,9 +52,23 @@ def destroy
       end
 
       def find_parent_config_template
-        @config_template = ConfigTemplate.authorized(:view_templates).find(params[:config_template_id])
-        not_found unless @config_template
-        @config_template
+        @config_template = ConfigTemplate.authorized(:view_templates).find(params[:config_template_id]) rescue nil
+      end
+
+      def find_hostgroup
+        Hostgroup.authorized(:view_templates).find(params[:hostgroup_id]) rescue nil
+      end
+
+      def find_environment
+        Environment.authorized(:view_templates).find(params[:environment_id]) rescue nil
+      end
+
+      def requested_resource
```

Comment:
Is it possible to use the `find_*_nested_object` support in the API base controller here?  It looks like you'd just specify allowed_nested_id and it could look up the base object.